### ISSUE TITLE
evaluate eval set every n iterations

### ISSUE THREAD

**Issue by zcyang** [microsoft/LightGBM#2885]
Hi,

I would like to track the training and validation loss in training. The only way I can do it is to provide train_data and valid_data in the eval set. However, since train_data is huge, adding train_data to the eval set significantly slows down the training. Is it possible to evaluate the eval set every n iterations. 

Or is there better way to do this? Thanks!

**Comment by guolinke** [comment]
metric_freq in cli.
